addons/mail/static/src/models/notification_group/notification_group.js

```diff
@@ -149,7 +164,13 @@ function factory(dependencies) {
             compute: '_computeThread',
         })
     };
-    NotificationGroup.identifyingFields = ['id'];
+    NotificationGroup.identifyingFields = ['res_model', 'res_id', 'notification_type'];
```

Comment:
Out of scope of PR, but I feel `res_model` and `res_id` are too technical and could be just a relational field to a python record... In discuss code it's somewhat `thread`...

Reply:
But also there's both python record and python model (with `res_id` being `null`). It's too much effort to introduce models for python model/record and dealing with relational fields for both models right now, so okay for `null` in time being (as said in another comment)